src/dmd/root/longdouble.d

```diff
@@ -324,7 +388,14 @@ void ld_setull(longdouble_soft* pthis, ulong d)
     d ^= (1L << 63);
     version(AsmX86)
     {
-        mixin(fld_local!("twoPow63"));
+        version(LDC)
+        {
+            // cannot access mutable global in pure function
+            static __gshared immutable _twoPow63 = longdouble_soft(1UL << 63, 0x3fff + 63);
```

Comment:
Just make the constants `const`. This also allows keeping the unittest at the bottom (but they need `@system` anyway).

For some reason, this needs an "offset" to be added for "fld_local" for D_InlineAsm_X86_64. "fld_local" is misnamed anyway, it should rather be called "fld_global".

Reply:
> Just make the constants const. 

I already tried that, but get this when making `twoPow63` const, with DMD 2.080 and `-m32mscoff`/`-m64`:

```
src\dmd\root\longdouble.d(756): Error: integer constant expression expected instead of longdouble_soft(9223372036854775808LU, cast(ushort)16446u)
src\dmd\root\longdouble.d-mixin-398(398): Error: bad type/size of operands lea
```

Reply:
Passed for x64 for me (but no longer now), but this seems to work for all cases:
```
        auto p = &twoPow63;
        mixin(fld_parg!("p"));
```
With this we can get rid of fld_local completely.

Reply:
Nice; reworked + squashed.